### PR TITLE
tikvclient: use global config instead write them again

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve? 
Unify config usage: reference global tidb config instead of use the global config to overwrite variables in modules.

Tests
Unit tests and Integration test.

### CODE REVIEW

File: store/tikv/client.go [hunk 1]
```diff
@@ -501,7 +490,8 @@ func (c *rpcClient) createConnArray(addr string) (*connArray, error) {
 	array, ok := c.conns[addr]
 	if !ok {
 		var err error
-		array, err = newConnArray(MaxConnectionCount, addr, c.security)
+		connCount := config.GetGlobalConfig().TiKVClient.GrpcConnectionCount
```
Comment (jackysp) [inline store/tikv/client.go]: `GrpcConnectionCount` could be 0?

Comment (siddontang) [reply store/tikv/client.go]: please check 0 here or in the global config initialization.

Comment (hicqu) [reply store/tikv/client.go]: Fixed.

**Pull request merged**